packages/admin-next/dashboard/src/routes/orders/order-create-refund/components/create-refund-form/create-refund-form.tsx

```diff
@@ -1,31 +1,40 @@
 import { zodResolver } from "@hookform/resolvers/zod"
 import { HttpTypes } from "@medusajs/types"
-import { Button, CurrencyInput, toast } from "@medusajs/ui"
+import { Button, CurrencyInput, Textarea, toast } from "@medusajs/ui"
 import { useForm } from "react-hook-form"
 import { useTranslation } from "react-i18next"
 import * as zod from "zod"
+import { upperCaseFirst } from "../../../../../../../../core/utils/src/common/upper-case-first"
 import { Form } from "../../../../../components/common/form"
+import { Combobox } from "../../../../../components/inputs/combobox"
 import { RouteDrawer, useRouteModal } from "../../../../../components/modals"
 import { useRefundPayment } from "../../../../../hooks/api"
 import { getCurrencySymbol } from "../../../../../lib/data/currencies"
 import { formatCurrency } from "../../../../../lib/format-currency"
 
 type CreateRefundFormProps = {
   payment: HttpTypes.AdminPayment
+  refundReasons: HttpTypes.AdminRefundReason[]
 }
 
 const CreateRefundSchema = zod.object({
   amount: zod.number(),
+  refund_reason_id: zod.string().nullish(),
+  note: zod.string().optional(),
 })
 
-export const CreateRefundForm = ({ payment }: CreateRefundFormProps) => {
+export const CreateRefundForm = ({
+  payment,
+  refundReasons,
+}: CreateRefundFormProps) => {
   const { t } = useTranslation()
   const { handleSuccess } = useRouteModal()
   const paymentAmount = payment.amount as unknown as number
 
   const form = useForm<zod.infer<typeof CreateRefundSchema>>({
     defaultValues: {
       amount: paymentAmount,
```

Comment:
suggestion: Default to 0 instead of the payment amount. On n+1 refunds, refunding the full amount won't be possible.